### PR TITLE
Add references to Pack Command Line docs, rename from CLI to Command Line

### PR DESCRIPTION
**Pull request by dfreilich**
* Add summaries to Pack Command Line index page, to make docs access clearer.
* Rename pack dir to pack-cli, to make the link clearer (for instance, https://buildpacks.io/docs/reference/pack/pack/ seems to be a mistake, rather than a clear link/breadcrumb)

Signed-off-by: David Freilich <dfreilich@vmware.com>

Closes https://github.com/buildpacks/pack/issues/812

image

### CODE REVIEW

File: content/docs/reference/pack-cli/_index.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
++++
+title="Pack Command Line Tool"
```
Comment (jromero) [inline content/docs/reference/pack-cli/_index.md:2]: From a marketing perspective this should be Pack CLI for consistency and simply sake.

```suggestion
title="Pack CLI"
```

**Comment by jromero** [comment]
A little bumbed that the summaries for pack subcommands don't have either short or long description presented. :(

File: content/docs/reference/pack-cli/_index.md (hunk 1 above)
Comment (dfreilich) [reply content/docs/reference/pack-cli/_index.md:2]: Ok. I had changed it due to the feedback from  buildpacks/pack#812, but I also find CLI a bit simpler

**Comment by dfreilich** [comment]
I know, and I agree. Without the `<!--more-->` tag, it was showing too much, and I couldn't find a way fo calling `.Short()`, so that we could have the short description before the `<!--more-->` tag in the index 😞

**Pull request merged**